sync/src/relayer/compact_block_process.rs

```diff
@@ -177,21 +176,28 @@ impl<'a, CS: ChainStore + 'static> CompactBlockProcess<'a, CS> {
                     return Ok(Status::AcceptBlock);
                 }
                 Err(missing) => {
-                    missing_indexes = missing;
+                    missing_indexes = missing.into_iter().map(|i| i as u32).collect::<Vec<_>>();
+
+                    // The missing_indexes should never be empty!
+                    if missing_indexes.is_empty() {
+                        debug_target!(
+                            crate::LOG_TARGET_RELAY,
+                            "Missing indexes should never be empty peer: {}, block: {}",
+                            self.peer,
+                            &block_hash,
+                        );
+                        return Ok(Status::MissingIndexesEmpty);
```

Comment:
`assert!(!missing_indexes.is_empty())` is better.

Reply:
Yes, we have talked about it before.